### DISCUSSION TITLE
ERROR window::os::wayland::copy_and_paste > Offer dnd_action None

### DISCUSSION THREAD

**Discussion by mckellygit** [wezterm/wezterm#1212] [Q&A] [answered]
Hi,
ty for wezterm.
I get this msg many, many times in my syslog -
```
ERROR window::os::wayland::copy_and_paste > Offer dnd_action None
```
But copy/paste seems to work fine for me.
1). what does it mean ?
2). can I stop it being reported or somehow fix it ?
thx.
-m

**Comment by mckellygit** [comment]
ok, I gather dnd means drag-and-drop, but I am just dragging a selection to copy it.
I use Ubuntu 18.04 with a fairly old mutter

**Reply by wez** [reply]
Could you expand on what you're doing to trigger this message?  Dragging from where to where?

**Reply by mckellygit** [reply]
Hi,
Its actually on all left releases.
Even if I double-click to highlight/select a word - when I release the left button the error msg appears.
So any left release after something is selected produces this msg.
I also see this sometimes -
```
Oct 11 08:35:42 wezterm.desktop[12540]:  2021-10-11T12:35:42.189Z ERROR window::os::wayland::window         > while sending paste to pipe: error writing to pipe: Broken pipe (os error 32)
```
but dont think that is related ?

**Reply by mckellygit** [reply]
ok, so if I select-drag whitespace - outside of any text - I do not get this error on the left release.
So the selection I think has to contain some chars.

**Reply by mckellygit** [reply]
I am using the nightly appimage, I can try to build it also, if I have a new enough dev env to do so.

**Comment by mckellygit** [comment]
```
            DataOfferEvent::Action { dnd_action } => {
                log::error!("Offer dnd_action {:?}", dnd_action);
            }
```
@wez is it an error that should be logged with action 'None' ?

**Comment by valpackett** [comment]
Drag&drop is not yet implemented. I might work on that in the near future.

**Reply by mckellygit** [reply]
ok, sure.  I am not trying to use Drag&drop, just selecting text.  Is this the expected behavior for all select [+ copy] ?
Its not a _big_ problem, but it is a syslog ERROR line for every selection I make.
I see it also from:
```
echo "foo" | xsel -i
```
(I haven't tried this with wl-copy yet).

**Comment by mckellygit** [comment]
I built wezterm from source and commented out this log line.
Thank you again for awesome wezterm :-)

**Reply by wez** [reply]
I'm still not clear why this only happens for you; is it specific to your Wayland compositor?  Which one are you using?

**Reply by mckellygit** [reply]
Yea, its probably just my env.  I'm on an up-to-date Ubuntu 18.04 with mutter 3.28.4 (gnome 3.28.2)

**Reply by mckellygit** [reply]
Just an update, I upgraded my CopyQ version recently, and I noticed I no longer see these log msgs on each selection.
thx, -m

**Comment by mckellygit** [comment, marked as answer]
So I think my issue was related to CopyQ.
